### ISSUE TITLE
[Android] executeSet breaks with empty array

### ISSUE THREAD

**Issue by patdx** [capacitor-community/sqlite#170]
**Describe the bug**

I am using some commands like:

```ts
CapacitorSQLite.executeSet({
  database: "my-db",
  set: [
    { statement: "DROP TABLE IF EXISTS download", values: [] },
    { statement: "CREATE TABLE download (src VARCHAR(255))", values: [] },
    { statement: "INSERT INTO download (src) values (?)", values: ["google.com"] },
  ],
});
```
This was causing an error message, after stepping through the code, it appears to have been caused at the following line in [Database.java](https://github.com/capacitor-community/sqlite/blob/b8ea80eee04f28159fc9c2a7e4b29bde138f170d/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java#L332):

```ts
Boolean isArray = _uSqlite.parse(values.get(0));
```

because `values.get(0)` is invalid, the values array has no items.

I tried to see if I could find a JavaScript workaround, like passing `values: [null]`, or `values: [[]]`, but these caused other strange errors instead.

I was finally able to fix the error by patching the Java source code:

```diff
- Boolean isArray = _uSqlite.parse(values.get(0));
+ Boolean isArray = values.size() > 0 ?  _uSqlite.parse(values.get(0)) : false;
```

**To Reproduce**

1. Open a capacitor android app with sqlite plugin and open the developer console
2. Enter some command like below in the developer console, replacing `my-db` with the name of a known database

```ts
Capacitor.Plugins.CapacitorSQLite.executeSet({
  database: "my-db",
  set: [
    { statement: "DROP TABLE IF EXISTS download", values: [] },
    { statement: "CREATE TABLE download (src VARCHAR(255))", values: [] },
    { statement: "INSERT INTO download (src) values (?)", values: ["google.com"] },
  ],
});
```

3. You should get a mysterious error like: `ExecuteSet: Index: 0, Size: 0`

**Expected behavior**

The command `executeSet` should work even with an empty values arrays. Users may want to combine value statements and no-value statements in a single transaction.

**Screenshots**

Here is a screenshot from debugging tools after replacing the problematic line:

![Screenshot 2021-09-14 193454](https://user-images.githubusercontent.com/353863/133245234-165d9c74-5956-4daa-9b5e-cfa3a4e5ad70.png)

**Smartphone (please complete the following information):**

- Device: Pixel 4 Api 30 Emulator
- OS: Android 11
- Browser Deault

**Context**

Actually I've playing with API using [https://www.npmjs.com/package/sql-template-tag](sql-template-tag), so my actual goal is to use some code like this:

```
import sql from "sql-template-tag";

const executeSet = (...statements) => {
  return CapacitorSQLite.executeSet({
    database: "my-db",
    set: statements.map((item) => ({
      statement: item.sql,
      values: item.values,
    })),
  });
};

executeSet(
  sql`DROP TABLE IF EXISTS download`,
  sql`CREATE TABLE download (src VARCHAR(255))`,
  sql`INSERT INTO download (src) values (${"google.com"})`
);
```

**Comment by jepiqueau** [comment]
@patdx thanks for using the plug-in excuteSet required to have values. To perform the Drop and CREATE TABLE ... use the command execute

**Comment by jepiqueau** [comment]
@patdx This has been fixed for all platforms in 3.2.2-2. I never thought to use the executeSet  like this. So thanks for your comments. Tell me if it works with `sql-template-tag` now and feel free to close the issue when tested in your side

**Comment by jepiqueau** [comment]
@patdx Not having news from you, i close the issue
